js/lykke2.js

```diff
@@ -0,0 +1,1174 @@
+        const price = this.safeString2 (trade, 'price', 'price');
+        let amount = this.safeString2 (trade, 'volume', 'amount');
+        if (amount === undefined) {
+            amount = this.safeString2 (trade, 'baseVolume', 'amount');
+        }
+        const side = this.safeStringLower (trade, 'side');
+        const fee = {
+            'cost': 0, // There are no fees for trading.
+            'currency': market['quote'],
+        };
+        return this.safeTrade ({
+            'id': id,
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': undefined,
+            'order': orderId,
+            'side': side,
+            'takerOrMaker': undefined,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': fee,
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (limit === undefined) {
+            limit = 100;
+        }
+        const request = {
+            'AssetPairId': market['id'],
+            // 'offset': 0,
+            'take': limit,
+        };
+        const response = await this.publicGetTradesPublicAssetPairId (this.extend (request, params));
+        const result = this.safeValue (response, 'payload', []);
+        //
+        //     {
+        //         "payload":[
+        //             {
+        //                 "id":"71df1f0c-be4e-4d45-b809-c108fad5f2a8",
+        //                 "assetPairId":"BTCUSD",
+        //                 "timestamp":1643345958414,
+        //                 "volume":0.00010996,
+        //                 "price":37205.723,
+        //                 "side":"buy"
+        //             }
+        //         ],
+        //         "error":null
+        //     }
+        //
+        return this.parseTrades (result, market, since, limit);
+    }
+
+    parseBalance (response) {
+        //
+        //     [
+        //         {
+        //             "assetId":"2a34d6a6-5839-40e5-836f-c1178fa09b89",
+        //             "available":0.1,
+        //             "reserved":0.0,
+        //             "timestamp":1644146723620
+        //         }
+        //     ]
+        //
+        const result = { 'info': response };
+        for (let i = 0; i < response.length; i++) {
+            const balance = response[i];
+            const currencyId = this.safeString (balance, 'assetId');
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            const free = this.safeString (balance, 'available');
+            const used = this.safeString (balance, 'reserved');
+            account['free'] = free;
+            account['used'] = used;
+            account['total'] = Precise.stringAdd (free, used);
```

Comment:
We don't need to add free + used here, cause it will be done in safeBalance. Let's remove line 639.